src/components/panGestureView/index.js

```diff
@@ -0,0 +1,170 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {PanResponder, Animated} from 'react-native';
+import {Constants} from '../../helpers';
+import {BaseComponent} from '../../commons';
+
+
+const DIRECTIONS = {
+  // VERTICAL
+  UP: 'up',
+  DOWN: 'down'
+};
+
+/**
+ * @description: PanGestureView component for drag and swipe gestures (supports only vertical gestures at the moment)
+ */
+export default class PanGestureView extends BaseComponent {
+  static displayName = 'PanGestureView'
+  static propTypes = {
+    /**
+     * onDismiss callback
+     */
+    onDismiss: PropTypes.func,
+    /**
+     * The direction of the allowed pan (default is down)
+     */
+    direction: PropTypes.oneOf(Object.values(DIRECTIONS))
+  };
+
+  static defaultProps = {
+    direction: DIRECTIONS.DOWN
+  };
+  
+  static directions = DIRECTIONS;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      deltaY: new Animated.Value(0)
+    };
+
+    this.panResponder = PanResponder.create({
+      onStartShouldSetPanResponder: this.handleStartShouldSetPanResponder,
+      onMoveShouldSetPanResponder: this.handleMoveShouldSetPanResponder,
+      onPanResponderGrant: this.handlePanResponderGrant,
+      onPanResponderMove: this.handlePanResponderMove,
+      onPanResponderRelease: this.handlePanResponderEnd,
+      onPanResponderTerminate: this.handlePanResponderEnd
+    });
+  }
+
+  handleStartShouldSetPanResponder = () => {
+    return true;
+  };
+  handleMoveShouldSetPanResponder = () => {
+    return true;
+  };
+  handlePanResponderGrant = () => {
+    this.swipe = false;
+  };
+  handlePanResponderMove = (e, gestureState) => {
+    const {direction} = this.getThemeProps();
+    let newValue = 0;
+    
+    // VERTICAL
+    const up = (direction === DIRECTIONS.UP);
+    const {deltaY} = this.state;
+
+    if (Math.abs(gestureState.vy) >= 1.8) {
+      if ((up && gestureState.vy < 0) || (!up && gestureState.vy > 0)) {
+        // Swipe
+        this.swipe = true;
+      }
+    } else if ((up && gestureState.dy < 0) || (!up && gestureState.dy > 0)) {
+      // Drag
+      newValue = gestureState.dy;
+      Animated.spring(deltaY, {
```

Comment:
`Animated.spring` being called twice almost the same, I would move to a function with a param to control the `toValue`